File: packages/js/src/core/codec/jsonCodec.ts
export function encodeJson(json: unknown): Uint8Array {
  if (typeof TextEncoder === "undefined") {
    throw new Error("TextEncoder is not available in this environment");
  }
  const encoder = new TextEncoder();
  const text = JSON.stringify(json);
  return encoder.encode(text);
}

export function decodeJson(bytes: Uint8Array): unknown {
  if (typeof TextDecoder === "undefined") {
    throw new Error("TextDecoder is not available in this environment");
  }
  const decoder = new TextDecoder("utf-8", { fatal: true });
  const text = decoder.decode(bytes);
  return JSON.parse(text);
}

File: packages/python/qraudio/codec/jsonCodec.py
from __future__ import annotations

import json


def encodeJson(value: object) -> bytes:
    text = json.dumps(value, separators=(",", ":"), ensure_ascii=False)
    return text.encode("utf-8")


def decodeJson(data: bytes) -> object:
    text = data.decode("utf-8", errors="strict")
    return json.loads(text)
